src/Audio.md

```diff
@@ -0,0 +1,103 @@
+
+Each channel generates an electronic signal; these signals are then mixed into two new channels (for stereo: one for the left ear, one for the right ear), which are then individually amplified, and then output either to the headphone jack, or the speaker[^speaker_mono].
+
+Channels 1 and 2, the "pulse channels", produce [pulse (rectangular) waves](https://en.wikipedia.org/wiki/Pulse_wave).
+Channel 3, the "wave" channel, produces user-defined waves.
+Channel 4 is the "[noise](https://en.wikipedia.org/wiki/Noise_in_music)" channel, producing a pseudo-random wave.
+
+The VIN channel is a signal received directly from the cartridge, allowing external hardware to supply a fifth sound channel.
+No licensed games used this feature, and it was omitted from the Game Boy Advance.
+
+::: tip POCKET MUSIC
+
+Despite rumors, *Pocket Music* does not use VIN. It refuses to run on the GBA for a different reason:
+the developer couldn't figure out how to silence buzzing associated with sample playback on the wave channel.
+
+:::
+
+[^speaker_mono]:
+The speaker merges back the two channels, losing the stereo aspect entirely.
+
+## Common concepts
+
+### APU
+
+The Game Boy's sound chip is called the <abbr title="Audio Processing Unit">APU</abbr>.
+
+The APU runs off the same master clock as the rest of the Game Boy, which is to say, it is fully synced with the CPU and [PPU](<#Rendering Overview>).
+This also means that the APU runs about 2.4% faster on the SGB1, increasing frequencies by as much and thus sounding slightly higher-pitched.
+The SGB2 does not suffer from this issue.
+
+The APU processes a lot of events at a fixed rate, which depends on the type of event.
+Thus, the APU interfaces use durations instead of frequencies, which may be confusing as signal theory and music are more typically based on the latter.
+Thus, durations will be expressed from their frequencies: for example, a "256 Hz tick" means "1/256th of a second".
+
+The length of ticks is not affected by [CGB double speed](<#FF4D - KEY1 - CGB Mode Only - Prepare Speed Switch>), so the APU works just the same regardless of CPU speed.
+
+::: warning
+
+The Game Boy's APU is actually full of tricky details; this chapter will mostly describe the intended / common behavior, and often paper over bugs & quirks.
+Readers wishing to learn more should read the [APU details](<#APU details>) chapter.
+
+:::
+
+### Envelope
+
+Each channel's [**envelope**](https://en.wikipedia.org/wiki/Envelope_(music)) can be configured.
+A signal's [envelope](https://en.wikipedia.org/wiki/Envelope_(waves)) is, roughly speaking, the variation of the signal's amplitude.
+On the GB, "envelope" basically designates the channel's loudness.
+
+All channels can have their envelope controlled to modulate their volume (CH3 less precisely than other channels).
+In addition, all channels but CH3 benefit from a functionality called **envelope sweep**, which automatically adjusts the envelope over time.
```

Comment:
Envelope is a modulation of amplitude, here a volume setting, so the terminology seems acceptable imo.

It's not possible to change this terminology, either, because it's used literally everywhere (including [hardware.inc](https://github.com/gbdev/hardware.inc/blob/05f5a9b6c7172abe1d7488080c1c050284c09226/hardware.inc#L230-L237)).

Reply:
The envelope is technically implemented as a sweep in the sense that it interpolates from one value to another. But hardware.inc, which copied the comment from nocash Pan Docs, seems to refer to the lower 3 bits of NRx2 as "Number of envelope sweep", and not the whole function of generating the envelope. In the overview, it was described like this with no mention of sweep (except referring to ch1's pitch sweep.)

>    Quadrangular wave patterns with sweep and envelope functions.
>    Quadrangular wave patterns with envelope functions.
>    Voluntary wave patterns from wave RAM.
>    White noise with an envelope function.

If anything, this seems like one of many quirks in Martin's English which should probably be improved instead of embraced.

Also note that nocash Pan Docs and hardware.inc don't refer to ch3 (`rNR32`) as having an envelope/being an envelope setting, but an output level.

Reply:
Updated.